rules/sort-variable-declarations.ts

```diff
@@ -89,16 +89,44 @@ export default createEslintRule<Options, MESSAGE_ID>({
           let dependencies: string[] = []
 
           let checkNode = (nodeValue: TSESTree.Node) => {
+            /**
+             * No need to check the body of functions and arrow functions
+             */
+            if (
+              nodeValue.type === 'ArrowFunctionExpression' ||
+              nodeValue.type === 'FunctionExpression'
+            ) {
+              return
+            }
+
             if (nodeValue.type === 'Identifier') {
               dependencies.push(nodeValue.name)
             }
 
+            if (nodeValue.type === 'Property') {
+              traverseNode(nodeValue.key)
+              traverseNode(nodeValue.value)
+            }
+
+            if (nodeValue.type === 'ConditionalExpression') {
+              traverseNode(nodeValue.test)
+              traverseNode(nodeValue.consequent)
+              traverseNode(nodeValue.alternate)
+            }
+
             if (
-              'body' in nodeValue &&
```

Comment:
This condition was removed because function expressions bodies are not required to have their dependencies before them:

```ts
let a = () => { return b},
     b = 1
```
is correct in Typescript.